################################################################################
SHORT DESCRIPTION: 
################################################################################
Add XEN_FBDEV2_FRONTEND to automatically select relevant kernel configuration
required for Xenfb2.

################################################################################
LONG DESCRIPTION: 
################################################################################

################################################################################
CHANGELOG 
################################################################################
Original Author: unknown
Port to 3.18: Eric Chanudet <chanudete@ainfosec.com>
Port to 4.14: Richard Turner <turnerr@ainfosec.com>
Port to 4.19: Richard Turner <turnerr@ainfosec.com>

################################################################################
REMOVAL 
################################################################################
When xenfb2 is not used anymore.

################################################################################
UPSTREAM PLAN
################################################################################
None this is specific to xenfb2 driver.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
Xenfb2.

################################################################################
PATCHES 
################################################################################
--- a/drivers/video/fbdev/Kconfig
+++ b/drivers/video/fbdev/Kconfig
@@ -2098,6 +2098,16 @@ config XEN_FBDEV_FRONTEND
 	  frame buffer driver.  It communicates with a back-end
 	  in another domain.
 
+config XEN_FBDEV2_FRONTEND
+        tristate "Xen virtual frame buffer support for XenClient"
+        depends on FB && XEN
+        select FB_CFB_COPYAREA
+        select FB_CFB_IMAGEBLIT
+        select FB_CFB_FILLRECT
+        default y
+        help
+          Hack to compile the needed functions for xenfb2
+
 config FB_METRONOME
 	tristate "E-Ink Metronome/8track controller support"
 	depends on FB
